db/schema_tables.cc

```diff
@@ -2025,6 +2025,28 @@ std::vector<shared_ptr<cql3::functions::user_function>> create_functions_from_sc
     return ret;
 }
 
+std::vector<shared_ptr<cql3::functions::user_aggregate>> create_aggregates_from_schema_partition(
+        replica::database& db, lw_shared_ptr<query::result_set> result, lw_shared_ptr<query::result_set> scylla_result) {
+    std::vector<shared_ptr<cql3::functions::user_aggregate>> ret;
+    for (const auto& row : result->rows()) {
+        auto agg_name = row.get_nonnull<sstring>("aggregate_name");
+        auto agg_args = read_arg_types(db, row, row.get_nonnull<sstring>("keyspace_name"));
+        const query::result_set_row *scylla_row_ptr = nullptr;
+        if (scylla_result) {
+            for (const auto& scylla_row : scylla_result->rows()) {
```

Comment:
This is a quadratic algorithm that we'll regret if we ever have a lot of aggregates. Wouldn't it be reasonable for a database that can have several hundreds of tables also to have a thousand aggregates?
Perhaps it's better to do one pass on scylla_result and prepare a hash table, and then do a second pass on result, and get a linear (actually, O(NlogN) is probably more accurate) algorithm?

Reply:
I also not a fan of quadratic algorithms, but I wonder if the extra complexity is worth it here. We perform the algorithm here only once during startup and the algorithm is only quadratic within a single keyspace_name.
But I think I'll try to implement it anyway, maybe it does not become particularly complicated

Reply:
I implemented the more efficient algorithm in the rebase. The complexity still isn't completely linear because we do iterate over all aggregates with the same name (and keyspace) to check if the argument types match, but we're already doing a similar procedure in a number of places when looking for UDFs, so this should be acceptable as well.